### PR TITLE
Add `ResizableThumbnailInterface`

### LINKED ISSUE #1861: Add interface for format resizers
## Feature Request

We are checking for the existance of a method on a class here: 

https://github.com/sonata-project/SonataMediaBundle/blob/3.x/src/DependencyInjection/Compiler/ThumbnailCompilerPass.php#L35

The only class that implement that method is:

https://github.com/sonata-project/SonataMediaBundle/blob/3.x/src/Thumbnail/FormatThumbnail.php#L46

We should really add an interface with the method and check that instead (adding the corresponding BC layers).
This issue becomes more problematic when trying to make this bundle compatible with php 8, because `is_callable` is not able to check with a string class name, but needs an object.

First saw in: https://github.com/sonata-project/SonataMediaBundle/pull/1859

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this is BC break.

Closes #1861 

## Changelog

```markdown
### Added
- Added `ResizableThumbnailInterface`
```

### CODE REVIEW

**Review by phansys — DISMISSED** [review]

File: src/Thumbnail/ResizableThumbnailInterface.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+namespace Sonata\MediaBundle\Thumbnail;
+
+use Sonata\MediaBundle\Resizer\ResizerInterface;
+
+interface ResizableThumbnailInterface
+{
+    public function addResizer(string $id, ResizerInterface $resizer): void;
+
+    /**
+     * @throws \LogicException if resizer is not found
```
Comment (VincentLanglet) [inline src/Thumbnail/ResizableThumbnailInterface.php:23]: Should we provide an `hasResizer` method then ?

File: src/Thumbnail/ResizableThumbnailInterface.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\MediaBundle\Thumbnail;
+
+use Sonata\MediaBundle\Resizer\ResizerInterface;
+
+interface ResizableThumbnailInterface
```
Comment (VincentLanglet) [inline src/Thumbnail/ResizableThumbnailInterface.php:18]: We should add the interface to `3.x` in order to allow people to implement it.

File: src/Thumbnail/ResizableThumbnailInterface.php (hunk 1 above)
Comment (jordisala1991) [reply src/Thumbnail/ResizableThumbnailInterface.php:25]: Added

**Review by VincentLanglet — APPROVED** [review]

**Comment by jordisala1991** [comment]
Fixed on the merge from 3.x #2071

**Pull request closed without merging**